test/unit-test/mqtt_agent_utest.c

```diff
@@ -853,3 +904,215 @@ void test_MQTTAgent_Terminate_success( void )
     TEST_ASSERT_NULL( command.pArgs );
     TEST_ASSERT_EQUAL_PTR( stubCompletionCallback, command.pCommandCompleteCallback );
 }
+
+/**
+ * @brief Test MQTTAgent_CommandLoop behaviour when sent command is NULL.
```

Comment:
Why would the sent command be NULL?
```suggestion
 * @brief Test MQTTAgent_CommandLoop behavior when sent command is NULL.
```

Reply:
This test is to cover the following branch else part
if( pCommand != NULL )
    {
        assert( pCommand->commandType < NUM_COMMANDS );
        commandFunction = pCommandFunctionTable[ pCommand->commandType ];
        pCommandArgs = pCommand->pArgs;
    }
    else
    {
        commandFunction = pCommandFunctionTable[ NONE ];
    }

Reply:
The name of this test could be better. That is the branch that it's testing but the reason it's NULL isn't that we sent NULL, there just wasn't a command to receive from the messaging interface.